File: widgets/selphid_plugin_v2/lib/SelphIDConfiguration.dart
import 'package:selphid_plugin/SelphIDDocumentType.dart';
import 'package:selphid_plugin/SelphIDTimeout.dart';
import 'SelphIDScanMode.dart';
import 'SelphIDCompressFormat.dart';

class SelphIDConfiguration{
  bool mDebug;
  bool mShowResultAfterCapture;
  bool mShowTutorial;
  SelphIDScanMode mScanMode;
  String mSpecificData;
  bool mFullscreen;
  double mTokenImageQuality;
  String mLocale;
  SelphIDDocumentType mDocumentType;
  SelphIDTimeout mTimeout;
  bool mEnableWidgetEventListener;
  bool mGenerateRawImages;
  SelphIDCompressFormat mCompressFormat;
  int mImageQuality;
  String mTranslationsContent;
  String mViewsContent;
  String mDocumentModels;
  int mCameraId;
  dynamic mParams;

  SelphIDConfiguration({
    this.mDebug = false,
    this.mShowResultAfterCapture = true,
    this.mShowTutorial = false,
    this.mScanMode = SelphIDScanMode.CAP_MODE_GENERIC,
    this.mSpecificData = "",
    this.mFullscreen = true,
    this.mLocale = "",
    this.mTokenImageQuality = 0.5,
    this.mDocumentType = SelphIDDocumentType.DT_IDCARD,
    this.mTimeout = SelphIDTimeout.T_SHORT,
    this.mEnableWidgetEventListener = false,
    this.mGenerateRawImages = false,
    this.mCompressFormat = SelphIDCompressFormat.T_JPEG,
    this.mImageQuality = 100,
    this.mViewsContent = "",
    this.mTranslationsContent = "",
    this.mDocumentModels = "",
    this.mCameraId = 100,
    this.mParams = "",
  });

  set params(dynamic params) {
    mParams = params;
  }

  dynamic get params {
    return mParams;
  }

  set cameraId(int camera) {
    mCameraId = camera;
  }

  int get cameraId {
    return mCameraId;
  }

  set documentModels(String documentModels) {
    mDocumentModels = documentModels;
  }

  String get documentModels {
    return mDocumentModels;
  }

  set viewsContent(String viewsContent) {
    mViewsContent = viewsContent;
  }

  String get viewsContent {
    return mViewsContent;
  }

  set translationsContent(String translationsContent) {
    mTranslationsContent = translationsContent;
  }

  String get translationsContent {
    return mTranslationsContent;
  }
  
  set imageQuality(int imageQuality) {
    mImageQuality = imageQuality;
  }

  int get imageQuality {
    return mImageQuality;
  }
  
  set compressFormat(SelphIDCompressFormat compressFormat) {
    mCompressFormat = compressFormat;
  }

  SelphIDCompressFormat get compressFormat {
    return mCompressFormat;
  }
  
  set generateRawImages(bool generateRawImages) {
    mGenerateRawImages = generateRawImages;
  }

  bool get generateRawImages {
    return mGenerateRawImages;
  }

  set enableWidgetEventListener(bool enableWidgetEventListener) {
    mEnableWidgetEventListener = enableWidgetEventListener;
  }

  bool get enableWidgetEventListener {
    return mEnableWidgetEventListener;
  }

  set debug(bool debug) {
    mDebug = debug;
  }

  bool get debug {
    return mDebug;
  }

  set documentType(SelphIDDocumentType documentType) {
    mDocumentType = documentType;
  }

  SelphIDDocumentType get documentType {
    return mDocumentType;
  }

  set showResultAfterCapture(bool showResultAfterCapture) {
    mShowResultAfterCapture = showResultAfterCapture;
  }

  bool get showResultAfterCapture {
    return mShowResultAfterCapture;
  }

  set showTutorial(bool showTutorial) {
    mShowTutorial = showTutorial;
  }

  bool get showTutorial {
    return mShowTutorial;
  }

  set tokenImageQuality(double tokenImageQuality) {
    mTokenImageQuality = tokenImageQuality;
  }

  double get tokenImageQuality {
    return mTokenImageQuality;
  }

  set scanMode(SelphIDScanMode scanMode) {
      mScanMode = scanMode;
  }

  SelphIDScanMode get scanMode {
    return mScanMode;
  }

  set timeout(SelphIDTimeout timeout) {
    mTimeout = timeout;
  }

  SelphIDTimeout get timeout {
    return mTimeout;
  }

  set specificData(String specificData) {
    mSpecificData = specificData;
  }

  String get specificData {
    return mSpecificData;
  }

  set fullscreen(bool fullscreen) {
    mFullscreen = fullscreen;
  }

  bool get fullscreen {
    return mFullscreen;
  }

  set locale(String locale) {
    mLocale = locale;
  }

  String get locale {
    return mLocale;
  }

  factory SelphIDConfiguration.fromJson(Map<String, dynamic> parsedJson){
    return SelphIDConfiguration(
        mDebug: parsedJson['debug'],
        mShowResultAfterCapture: parsedJson['showResultAfterCapture'],
        mShowTutorial: parsedJson['showTutorial'],
        mScanMode: parsedJson['scanMode'],
        mDocumentType: parsedJson['documentType'],
        mSpecificData: parsedJson['specificData'],
        mFullscreen: parsedJson['fullscreen'],
        mLocale: parsedJson['locale'],
	      mTokenImageQuality: parsedJson['tokenImageQuality'],
        mTimeout: parsedJson['timeout'],
        mEnableWidgetEventListener: parsedJson['enableWidgetEventListener'],
        mGenerateRawImages: parsedJson['generateRawImages'],
        mCompressFormat: parsedJson['compressFormat'],
        mImageQuality: parsedJson['imageQuality'],
        mViewsContent: parsedJson['viewsContent'],
        mTranslationsContent: parsedJson['translationsContent'],
        mDocumentModels: parsedJson['documentModels'],
        mCameraId: parsedJson['cameraId'],
        mParams: parsedJson['params'],
    );
  }

  Map<String, dynamic> toJson() =>
      {
        'debug': mDebug,
        'showResultAfterCapture': mShowResultAfterCapture,
        'showTutorial': mShowTutorial,
        'scanMode': mScanMode.toString(),
        'documentType': mDocumentType.toString(),
        'specificData': mSpecificData,
        'fullscreen': mFullscreen,
        'locale': mLocale,
        'tokenImageQuality': mTokenImageQuality,
        'timeout': mTimeout.toString(),
        'enableWidgetEventListener': mEnableWidgetEventListener,
        'generateRawImages': mGenerateRawImages,
        'compressFormat': mCompressFormat.toString(),
        'imageQuality': mImageQuality,
        'viewsContent': mViewsContent,
        'translationsContent': mTranslationsContent,
        'documentModels': mDocumentModels,
        'cameraId': mCameraId,
        'params': mParams,
      };
}

File: Sources/facephiSDK-ios-package/Request/SelphIdConfiguration.swift
//
//  SelphIdConfigurationData.swift
//  ios
//
//  Created by Faustino Flores García on 1/12/21.
//

import Foundation

public enum SelphIDDocumentType: String {
    case DT_IDCARD = "DT_IDCard"
    case DT_PASSPORT = "DT_Passport"
    case DT_DRIVERSLICENSE = "DT_DriversLicense"
    case DT_FOREIGNCARD = "DT_ForeignCard"
    case DT_CREDITCARD = "DT_CreditCard"
    case DT_CUSTOM = "DT_Custom"
}

public enum SelphIDTimeout: String {
    case T_SHORT = "Short"
    case T_MEDIUM = "Medium"
    case T_LONG = "Long"
}

public enum SelphIDScanMode: String {
    case CAP_MODE_GENERIC = "GenericMode"
    case CAP_MODE_SPECIFIC = "SpecificMode"
    case CAP_MODE_SEARCH = "SearchMode"
}

public enum SelphIDCompressFormat: String {
    case T_JPEG = "jpeg"
    case T_PNG = "png"
}

public struct SelphIdConfiguration: Codable {
    var debug = false
    var showResultAfterCapture = true
    var showTutorial = false
    var scanMode = SelphIDScanMode.CAP_MODE_SEARCH.rawValue
    var specificData = ""
    var fullscreen = true
    var tokenImageQuality = 0.5
    var locale = ""
    public var documentType = SelphIDDocumentType.DT_IDCARD.rawValue
    var timeout = SelphIDTimeout.T_SHORT.rawValue
    var enableWidgetEventListener = false
    var generateRawImages = false
    var compressFormat = SelphIDCompressFormat.T_JPEG.rawValue
    var imageQuality = 100
    var translationsContent = ""
    var viewsContent = ""
    var documentModels = ""
    public var license = ""

    public init(debug: Bool,
                showResultAfterCapture: Bool,
                showTutorial: Bool,
                scanMode: String,
                specificData: String,
                fullscreen: Bool,
                tokenImageQuality: Double,
                locale: String,
                documentType: String,
                timeout: String,
                enableWidgetEventListener: Bool,
                generateRawImages: Bool,
                compressFormat: String,
                imageQuality: Int,
                translationsContent: String,
                viewsContent: String,
                documentModels: String,
                license: String)
    {
        self.debug = debug
        self.showResultAfterCapture = showResultAfterCapture
        self.showTutorial = showTutorial
        self.scanMode = scanMode
        self.specificData = specificData
        self.fullscreen = fullscreen
        self.tokenImageQuality = tokenImageQuality
        self.locale = locale
        self.documentType = documentType
        self.timeout = timeout
        self.enableWidgetEventListener = enableWidgetEventListener
        self.generateRawImages = generateRawImages
        self.compressFormat = compressFormat
        self.imageQuality = imageQuality
        self.translationsContent = translationsContent
        self.viewsContent = viewsContent
        self.documentModels = documentModels
        self.license = license
    }

    public init() {}
}
